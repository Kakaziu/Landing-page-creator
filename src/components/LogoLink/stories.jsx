import LogoLink from '.';

export default {
  title: 'LogoLink',
  component: LogoLink,
  args: {
    text: 'LogoLink',
    srcImg: './assets/images/logo.webp',
    link: 'http://localhost',
  },
};

export const ImageOnly = (args) => (
  <div>
    <LogoLink {...args} />
  </div>
);

export const TextOnly = (args) => (
  <div>
    <LogoLink {...args} />
  </div>
);

TextOnly.args = {
  srcImg: '',
};
