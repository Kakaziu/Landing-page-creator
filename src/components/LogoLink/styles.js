import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.primaryColor};
    text-decoration: none;
    font-size: ${theme.fonts.sizes.medium};
    display: flex;
    align-items: center;

    > img {
      max-width: 15rem;
      max-height: 8rem;
    }
  `}
`;
