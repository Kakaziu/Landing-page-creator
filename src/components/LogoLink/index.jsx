import P from 'prop-types';
import * as Styled from './styles';
import Heading from '../Heading';

const LogoLink = ({ text, srcImg, link }) => {
  return (
    <Heading>
      <Styled.Container href={link}>
        {srcImg ? <img src={srcImg} alt={text} /> : <span>{text}</span>}
      </Styled.Container>
    </Heading>
  );
};

LogoLink.propTypes = {
  text: P.string.isRequired,
  srcImg: P.string,
  link: P.string.isRequired,
};

export default LogoLink;
