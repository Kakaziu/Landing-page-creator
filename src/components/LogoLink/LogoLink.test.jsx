import { screen } from '@testing-library/react';
import LogoLink from '.';
import renderTheme from '../../styles/render-theme';

describe('<LogoLink/>', () => {
  it('should render no image logo', () => {
    renderTheme(<LogoLink text="Logo" link="#" />);

    const logo = screen.getByText('Logo');
    expect(logo).toBeInTheDocument();
  });

  it('should render with image', () => {
    renderTheme(
      <LogoLink text="Logo" link="#" srcImg="./assets/images/logo.webp" />,
    );

    const logo = screen.getByAltText('Logo');
    expect(logo).toBeInTheDocument();
  });

  it('should render with correct link', () => {
    renderTheme(
      <LogoLink text="Logo" link="#" srcImg="./assets/images/logo.webp" />,
    );

    const logo = screen.getByAltText('Logo');
    expect(logo.parentElement).toHaveAttribute('href', '#');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink text="LogoLink" link="#target" />,
    );

    expect(container).toMatchSnapshot();
  });
});
