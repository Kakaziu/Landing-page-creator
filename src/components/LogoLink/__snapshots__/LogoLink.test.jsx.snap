// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink/> should match snapshot 1`] = `
.c1 {
  color: #0A1128;
  text-decoration: none;
  font-size: 2.4rem;
  display: flex;
  align-items: center;
}

.c1 >img {
  width: 20rem;
}

.c0 {
  color: #0A1128;
  text-transform: none;
  font-size: 3.2rem;
}

<div>
  <h1
    class="c0"
  >
    <a
      class="c1"
      href="#target"
    >
      <span>
        LogoLink
      </span>
    </a>
  </h1>
</div>
`;
